### PR TITLE
nmcli: manage dummy connections

### PR DESCRIPTION
**Pull request by rekup**
##### SUMMARY
Add ability to manage dummy interfaces using the `nmcli` module

Fixes https://github.com/ansible-collections/community.general/issues/724

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
nmcli

##### ADDITIONAL INFORMATION
- add ability to manage dummy interfaces

### CODE REVIEW

**Comment by felixfontein** [comment]
Please note that the tests currently have a conflict due to #3081 being merged.

CC @Ajpantuso @hummeltech maybe this PR is of interest to you as well.

File: changelogs/fragments/3132-nmcli-dummy.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - nmcli - add dummy interface support (https://github.com/ansible-collections/community.general/issues/724).
```
Comment (Ajpantuso) [inline changelogs/fragments/3132-nmcli-dummy.yaml:2]:
```suggestion
  - nmcli - add ``dummy`` interface support (https://github.com/ansible-collections/community.general/issues/724).
```

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
The test conflicts are just because the additions occupy the same space. I would pull out the tests for this PR to a scratch file, rebase, and then add the tests back. Otherwise looks okay to me.

**Review by hummeltech — APPROVED** [review]
Looks good!

**Review by Ajpantuso — APPROVED** [review]
LGTM, thanks!

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@rekup thanks for contributing this!
@Ajpantuso @hummeltech thanks for reviewing!

**Pull request merged**